### PR TITLE
apply lockout for non fictional switch

### PR DESCRIPTION
**Pull request by souissimai**

### CODE REVIEW

**Review by antoinebhs — COMMENTED** [review]
I'm not sure we want to do the change in this service. Don't we want to change it in network-store in the default method? 
It seems very strange (wrong?) to have the two types of disconnect mixed together inside the same modification:
https://github.com/gridsuite/network-modification-server/blob/ef95cb67522525088a0167679cb2604bc298b420/src/main/java/org/gridsuite/modification/server/modifications/OperatingStatusModification.java#L146-L148 
vs 
https://github.com/gridsuite/network-modification-server/blob/ef95cb67522525088a0167679cb2604bc298b420/src/main/java/org/gridsuite/modification/server/modifications/OperatingStatusModification.java#L162

Also, I thought we just needed to ignore fictional switch, the default was IS_CLOSED_BREAKER so I would have chosen IS_NONFICTIONAL_CLOSED_BREAKER, why IS_NONFICTIONAL_BREAKER? 

I also don't really understand the changes in the tests.

(we could also fix the sonar issue but as I'm not sure it's the right solution, it's not a big deal)

cc @SlimaneAmar

File: src/main/java/org/gridsuite/modification/server/modifications/OperatingStatusModification.java [hunk 1]
```diff
@@ -158,13 +159,17 @@ private void applyEnergiseEquipmentEnd(ReportNode subReportNode, Identifiable<?>
...
     private boolean disconnectAllTerminals(Identifiable<?> equipment) {
-        return ModificationUtils.getInstance().getTerminalsFromIdentifiable(equipment).stream().allMatch(this::disconnectOneTerminal);
+        return ModificationUtils.getInstance().getTerminalsFromIdentifiable(equipment).stream().allMatch(this::disconnectNonFictionalOneTerminal);
     }
 
     private boolean disconnectOneTerminal(Terminal terminal) {
         return !terminal.isConnected() || terminal.disconnect();
     }
 
+    private boolean disconnectNonFictionalOneTerminal(Terminal terminal) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/modifications/OperatingStatusModification.java:169]: Remove `NonFictional` in the name and pass the predicate in argument

File: src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationLockoutLineTest.java [hunk 2]
```diff
@@ -79,35 +80,53 @@ protected void assertAfterNetworkModificationDeletion() {
...
+                .setVoltageLevel1("v1f")
+                .setVoltageLevel2("v3f")
+                .setNode1(0)
+                .setNode2(0)
+                .setX(12)
+                .setR(7)
+                .add();
+        assertNotNull(line);
+
+        line.getTerminal("v1f").disconnect(SwitchPredicates.IS_NONFICTIONAL);
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationLockoutLineTest.java:103]: Don't understand !
Need to create a fictional switch and test disconnect
And another tests each with a type of switch ?

File: src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationLockoutLineTest.java [hunk 3]
```diff
@@ -79,35 +80,53 @@ protected void assertAfterNetworkModificationDeletion() {
...
-
-        // modification action empty
-        modificationInfos.setEquipmentId("line2");
-        modificationInfos.setAction(null);
-        modificationJson = mapper.writeValueAsString(modificationInfos);
-        mockMvc.perform(post(getNetworkModificationUri()).content(modificationJson).contentType(MediaType.APPLICATION_JSON))
-            .andExpectAll(
-                    status().isBadRequest(),
-                    content().string(new NetworkModificationException(OPERATING_ACTION_TYPE_EMPTY).getMessage())
-            );
```
Comment (antoinebhs) [inline src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationLockoutLineTest.java:102]: Put back this test?

Comment (antoinebhs) [reply src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationLockoutLineTest.java:102]: Add:
* one test with a fictitious switch => it should throw an exception
* one test with a non breaker switch

File: src/main/java/org/gridsuite/modification/server/modifications/OperatingStatusModification.java (hunk 1 above)
Comment (souissimai) [reply src/main/java/org/gridsuite/modification/server/modifications/OperatingStatusModification.java]: done

File: src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationLockoutLineTest.java (hunk 3 above)
Comment (souissimai) [reply src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationLockoutLineTest.java:102]: done

File: src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationLockoutLineTest.java [hunk 4]
```diff
@@ -79,14 +79,25 @@ protected void assertAfterNetworkModificationDeletion() {
...
+    @Test
+    public void testLockoutModification() throws Exception {
+        OperatingStatusModificationInfos modificationInfos = (OperatingStatusModificationInfos) buildModification();
+        modificationInfos.setEquipmentId("line2");
+        modificationInfos.setAction(OperatingStatusModificationInfos.ActionType.LOCKOUT);
+        String modificationJson = mapper.writeValueAsString(modificationInfos);
+        mockMvc.perform(post(getNetworkModificationUri()).content(modificationJson).contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+        assertAfterNetworkModificationCreation();
+    }
```
Comment (antoinebhs) [inline src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationLockoutLineTest.java:91]: Isn't already tested in the testCreate()

File: src/test/java/org/gridsuite/modification/server/utils/NetworkCreation.java [hunk 5]
```diff
@@ -27,6 +27,41 @@ public static Network create(UUID uuid, boolean createHvdcLine) {
         return create(uuid, createHvdcLine, new NetworkFactoryImpl());
     }
 
+    public static Network createSwitchNetwork(UUID uuid, NetworkFactory networkFactory) {
```
Comment (antoinebhs) [inline src/test/java/org/gridsuite/modification/server/utils/NetworkCreation.java:30]: Add a comment to explain what's the point of this network: fictitious switch and different switch kind

File: src/test/java/org/gridsuite/modification/server/utils/NetworkCreation.java [hunk 6]
```diff
@@ -27,6 +27,41 @@ public static Network create(UUID uuid, boolean createHvdcLine) {
...
+        createSwitch(vl3, "br7", "br7", SwitchKind.BREAKER, false, false, false, 1, 2);
+        createSwitch(vl3, "br8", "br8", SwitchKind.BREAKER, false, false, false, 1, 3);
+        createLoad(vl3, "l1", "l1", 0, 180, 18.0, "cn0", 3, ConnectablePosition.Direction.TOP);
+
+        createLine(network, "line1", "line1", "vl1", "vl2", 2, 0, 0.1, 10.0, 0.0, 0.0, 0.0, 0.0, "line1", 1, ConnectablePosition.Direction.TOP, "line1", 1, ConnectablePosition.Direction.TOP);
+        createLine(network, "line2", "line2", "vl2", "vl3", 2, 2, 0.1, 10.0, 0.0, 0.0, 0.0, 0.0, "line2", 1, ConnectablePosition.Direction.TOP, "line2", 1, ConnectablePosition.Direction.TOP);
+        createLine(network, "line3", "line3", "vl1", "vl3", 3, 3, 0.1, 10.0, 0.0, 0.0, 0.0, 0.0, "line3", 1, ConnectablePosition.Direction.TOP, "line3", 1, ConnectablePosition.Direction.TOP);
+
+        network.getVariantManager().setWorkingVariant(VariantManagerConstants.INITIAL_VARIANT_ID);
+        network.getVariantManager().cloneVariant(VariantManagerConstants.INITIAL_VARIANT_ID, VARIANT_ID);
```
Comment (antoinebhs) [inline src/test/java/org/gridsuite/modification/server/utils/NetworkCreation.java:60]: Why do we need to clone the variant?

File: src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationLockoutLineTest.java [hunk 7]
```diff
@@ -73,57 +74,77 @@ protected void assertAfterNetworkModificationCreation() {
...
     @Override
     protected void assertAfterNetworkModificationDeletion() {
         // go back to init status
         TestUtils.assertOperatingStatus(getNetwork(), TARGET_LINE_ID, OTHER_BRANCH_STATUS);
     }
 
+    @Test
+    public void testLockoutLinesWithLoadBreakerSwitches() throws Exception {
+        //Lockout line with switches of kind LOAD_BREAK_SWITCH
+        getNetwork().getSwitch("br11").setProperty("Kind", SwitchKind.LOAD_BREAK_SWITCH.toString());
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationLockoutLineTest.java:98]: A property does not modify the type !

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationLockoutLineTest.java [hunk 8]
```diff
@@ -96,34 +148,24 @@ public void testCreateWithErrors() throws Exception {
...
-        modificationInfos.setEquipmentId("cantdisconnect");
+        // Add a line that can't be disconnected : with fictitious switches
+        Line line1 = createLineAndSwitches(SwitchKind.BREAKER, true);
+        assertNotNull(line1);
+        modificationInfos.setEquipmentId("line1");
         modificationInfos.setAction(OperatingStatusModificationInfos.ActionType.LOCKOUT);
         modificationJson = mapper.writeValueAsString(modificationInfos);
         mockMvc.perform(post(getNetworkModificationUri()).content(modificationJson).contentType(MediaType.APPLICATION_JSON))
                 .andExpect(status().isOk());
-        assertNull(getNetwork().getLine("cantdisconnect").getExtension(OperatingStatus.class));
```
Comment (antoinebhs) [inline src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationLockoutLineTest.java:126]: Why remove ? 
`        assertNull(line1.getExtension(OperatingStatus.class));`
?

File: src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationLockoutLineTest.java [hunk 9]
```diff
@@ -73,20 +99,46 @@ protected void assertAfterNetworkModificationCreation() {
         TestUtils.assertOperatingStatus(getNetwork(), TARGET_LINE_ID, TARGET_BRANCH_STATUS);
     }
 
+    private void assertAfterNetworkModificationLockout(String lineID) throws Exception {
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationLockoutLineTest.java:102]: Rename by `testLockoutLine`

**Comment by SlimaneAmar** [comment]
Remove the Sonar issue

File: src/test/java/org/gridsuite/modification/server/utils/NetworkCreation.java (hunk 6 above)
Comment (souissimai) [reply src/test/java/org/gridsuite/modification/server/utils/NetworkCreation.java:72]: removed

**Review by antoinebhs — APPROVED** [review]

**Pull request merged**